File: 1002-find-common-characters/1002-find-common-characters.cpp
class Solution {
public:
    vector<string> commonChars(vector<string>& words) {

       vector<string> res;
       for (char c : words[0]) {
           if (all_of(words.begin(), words.end(), [c](const string& word) { return word.find(c) != string::npos; })) {
               res.push_back(string(1, c));
               for (int j = 1; j < words.size(); ++j) {
                   size_t pos = words[j].find(c);
                   if (pos != string::npos) {
                       words[j].erase(pos, 1);
                   }
               }
           }
       }
         return res; 
        
    }
};

File: 1002-find-common-characters/1002-find-common-characters.py
class Solution(object):
    def commonChars(self, words):
        """
        :type words: List[str]
        :rtype: List[str]
        """
        res = []
        for i in words[0]:
            if all(i in word for word in words):
                res.append(i)
                for j in range(1, len(words)):
                    words[j] = words[j].replace(i, "", 1)
        return res